packages/gestalt/src/Masonry.js

```diff
@@ -322,22 +328,14 @@ export default class Masonry<T> extends React.Component<Props<T>, State<T>> {
     return null;
   }
 
-  setGridWrapperRef = (ref: ?HTMLElement) => {
-    this.gridWrapper = ref;
-  };
-
-  setScrollContainerRef = (ref: ?ScrollContainer) => {
-    this.scrollContainer = ref;
-  };
-
   fetchMore = () => {
-    const { loadItems } = this.props;
+    const { loadItems, items } = this.props;
     if (loadItems && typeof loadItems === 'function') {
       this.setState(
         {
           isFetching: true,
         },
-        () => loadItems({ from: this.props.items.length })
+        () => loadItems({ from: items.length })
```

Comment:
is this change ok? the callback here is executed asynchronously after the state update but by pulling out `items` into a variable in this closure we are setting a pointer to it before the function is executed. tl;dr would `this.props.items` ever change reference by the time its used inside this set state callback?

Reply:
I waffled on this a bit when I wrote it, but I'm pretty sure it's right. Not only is it cosmetically better, but it actually conveys the intent better as well, and I'm pretty confident that the edge case would never be exposed in our codebase, or if it was, it would have no adverse effects.

Reply:
👍